Modules/install-modules.R.in

```diff
@@ -0,0 +1,145 @@
+LOCAL_MODULES_DIR <- file.path("@PROJECT_SOURCE_DIR@", "Modules", "local")
+BUNDLE_DOWNLOAD_DIR <- file.path("@PROJECT_BINARY_DIR@", "Modules", "downloads")
+
+if(@IS_FLATPAK_USED@) {
+    print("Linking to pre-build modules dir")
+    unlink("@MODULES_BINARY_PATH@", recursive=TRUE)
+    file.symlink("/app/Modules/", "@MODULES_BINARY_PATH@")
+}
+
+getPlatform <- function() {
+    info <- Sys.info()
+    os <- ""
+    switch(info[['sysname']],
+        Windows = {os <- "Windows-x86_64"},
+        Linux   = {os <- "Linux"},
+        Darwin  = {os <- if(info[['machine']] == "arm64") "MacOS-arm64" else "MacOS-x86_64"}
+    )
+    os
+}
+mkdir <- function(paths) {
+  for (path in paths)
+    if (!dir.exists(path))
+      dir.create(path, recursive = TRUE)
+}
+mkdir(c(MODULES_DIR, LOCAL_MODULES_DIR, BUNDLE_DOWNLOAD_DIR))
+
+
+.libPaths("@JASP_MODULE_BUNDLE_MANAGER_LIBRARY@")
+library("jaspModuleBundleManager")
+library("tools")
+
+#check which bundles need downloading
+remoteBundles <- rjson::fromJSON(file="@MODULES_SOURCE_PATH@/remote-bundles.json")[[getPlatform()]]
+remoteBundleURLS <- unlist(sapply(remoteBundles, function(x){x$url}))
+
+remoteBundleDownloaded <- c()
+if(file.exists("@MODULES_BINARY_PATH@/bundles-downloaded.txt")) {
+    tryCatch({
+        remoteBundleDownloaded <- read.delim("@MODULES_BINARY_PATH@/bundles-downloaded.txt", header = FALSE)[[1]]
+        remoteBundleDownloaded <- remoteBundleDownloaded[file.exists(file.path(BUNDLE_DOWNLOAD_DIR, basename(remoteBundleDownloaded)))]
+    }, error = function(e) { remoteBundleDownloaded <- c() })
+}
+toDownload <- remoteBundles[!remoteBundleURLS %in% remoteBundleDownloaded]
+
+#download and write a file with all our successes
+downloaded <- c()
+if(length(toDownload)) {
+    cat("We will download: ", paste(toDownload, sep=' '), '\n')
+    download <- function(bundle) {
+        req <- tryCatch({
+            path <- file.path(BUNDLE_DOWNLOAD_DIR, basename(bundle$url))
+            res <- curl::curl_fetch_disk(bundle$url, path)
+            if(sha256sum(files=path) != bundle$checksum) {
+                file.remove(path)
+                warning(paste0("Checksum of: ", bundle$url, "Does not match. Skipping"))
+                res <- list(status_code=404)
+            }
+            res 
+        }, error = function(e) { print(e); list(status_code=404) })
+        if(req$status_code != 200)
+            return(FALSE)
+        TRUE
+    }
+    res <- sapply(toDownload, download)
+    success <- unlist(sapply(toDownload[res], function(x){x$url}))
+    downloaded <- unique(c(remoteBundleDownloaded[remoteBundleDownloaded %in% remoteBundleURLS], success))
+    write(downloaded, "@MODULES_BINARY_PATH@/bundles-downloaded.txt")
+
+    if(!all(res)) {
+        toDownload <- toDownload[!res]
+        cat("Unable to download the following bundles: ", paste(toDownload, sep = ' '), '\n')
+    }
+
+    #clean up the download folder a bit by removing all bundles we where not instructed to download
```

Comment:
Is this to clean up older previously installed modules?

Reply:
yeah previous downloaded modules for the sake of space. Not critical.